miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -374,6 +382,10 @@ end
 #!
 #! - ASSET is the asset to remove from the vault.
 export.account_vault_remove_asset
+    # TODO: we execute `push.0 drop` before `emit` as decorators are not supported without other
+    # instructions - see: https://github.com/0xPolygonMiden/miden-vm/issues/1122
+    push.0 drop emit.ACCOUNT_VAULT_BEFORE_REMOVE_ASSET_EVENT
```

Comment:
shouldn't we be consistent on emitting these events after the `authenticate_account_origin`?

Reply:
Ah, indeed, let me fix this